keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,1235 @@
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+- Some CPU Processor vendors have a multi-chiplet architecture consisting of multiple L3 cache groups. Similar to how certain workloads benefit when deployed using the Static CPU Manager Policy option, some workloads see an additional performance uplift when affined to a single L3 cache or distrubted across fewer L3 caches.
+- The existing Static CPU policy assigns CPUs in a numerical order. The user's pod CPU requirements can vary and won't always be a perfect fit to the CPUs tied to the same L3 cache group.
+- With an uncore cache alignment feature enabled in the static policy, the user can depend on the CPU scheduler to optimally assign CPUs with regards to minimizing distribution across L3 caches. 
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- Feature is enabled by a static policy option flag.  
+- It does not change behavior of non-static policy
+- It preserves the behavior of other static options.
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+- Add `UnCoreCacheID` in `CPUInfo` and modify CPUTopology struct to also track uncorecaches in `topology.go`
+  
+  ```go
+  // CPUTopology contains details of node cpu, where :
+  // CPU  - logical CPU, cadvisor - thread
+  // Core - physical CPU, cadvisor - Core
+  // Socket - socket, cadvisor - Socket
+  // NUMA Node - NUMA cell, cadvisor - Node
+  // UnCoreCacheID - Split L3 Cache Topology, cadvisor
+  type CPUTopology struct {
+  NumCPUs        int
+  NumCores       int
+  NumUnCoreCache int
+  NumSockets     int
+  NumNUMANodes   int
+  CPUDetails     CPUDetails
+  }
+  type CPUInfo struct {
+    NUMANodeID int
+    SocketID   int
+    CoreID     int
+    UnCoreCacheID  int
+  }
+  ```
+  
+- Get the UnCoreCacheID for each CPU from cadvisor.
+
+  ```go
+  func Discover(machineInfo *cadvisorapi.MachineInfo) (*CPUTopology, error) {
+  CPUDetails[cpu] = CPUInfo{
+    CoreID:     coreID,
+    SocketID:   core.SocketID,
+    NUMANodeID: node.Id,
+    UnCoreCacheID:  core.UncoreCaches[0].Id
+    }
+  }
+    ```
+
+- Setting the `CPUManagerPolicy` to `static` followed by adding a proposed option in `CPUManagerPolicyOptions` called `align-cpus-by-uncorecache` and setting it to `true`.
```

Comment:
Bools are difficult to extend. We sometimes use enums for this reason. I would prefer an enum especially if this needs to be extended.

Reply:
@ffromani wdyt?

Reply:
I think this fits the direction we took so far https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/policy_options.go#L92

the cpumanager options are more akin to feature toggles (on/off) atm. I think this is a valid concern but is a more general one which would require a separate conversation.

Reply:
Thank you for clearing that up. Do we have a list of allowed values for that map?

Reply:
atm the string values are intepreted per-option so, basically there's no further validation. It's really a raw string which is `ParseBool`ed when we process the options. The silver lining is this is very easy to extend in the future if we need more values than "true/false".

Reply:
Is it safe to resolve this, I think we're in a agreement here

Reply:
agreed